features/mbedtls/targets/TARGET_Silicon_Labs/crypto_ecp.c

```diff
@@ -0,0 +1,1732 @@
+                                         CRYPTO_CMD_INSTR_DDATA0TODDATA2
+                                         );
+                    }
+
+                    CRYPTO_DDATA0_260_BITS_WRITE(crypto, D);
+                    crypto->CMD = CRYPTO_CMD_INSTR_DDATA0TODDATA3;
+
+                } /* if (k == -1) */
+            }
+        } /* else: !if((C[31:0] & 0x1) == 0x0) */
+
+        crypto->CMD = CRYPTO_CMD_INSTR_DDATA2TODDATA0;
+
+        lsb_U = CRYPTO_DData0_4LSBitsRead(crypto);
+
+        /* if ((U[31:0] & 0x1) == 0x1) */
+        if((lsb_U & 0x1) == 0x1)
+        {
+            CRYPTO_EXECUTE_3( crypto,
+                              CRYPTO_CMD_INSTR_SELDDATA2DDATA2,
+                              CRYPTO_CMD_INSTR_SHRA,
+                              CRYPTO_CMD_INSTR_DDATA0TODDATA2);
+
+            CORE_ENTER_CRITICAL();
+            CRYPTO_DDataWrite(&crypto->DDATA0, N);
+            CORE_EXIT_CRITICAL();
+
+            CRYPTO_EXECUTE_6( crypto,
+                              CRYPTO_CMD_INSTR_SELDDATA0DDATA0,
+                              CRYPTO_CMD_INSTR_SHR,
+                              CRYPTO_CMD_INSTR_SELDDATA0DDATA2,
+                              CRYPTO_CMD_INSTR_CSET,
+                              CRYPTO_CMD_INSTR_ADDC,
+                              CRYPTO_CMD_INSTR_DDATA0TODDATA2);
+        }
+        else
+        {
+            CRYPTO_EXECUTE_3(crypto,
+                             CRYPTO_CMD_INSTR_SELDDATA2DDATA2,
+                             CRYPTO_CMD_INSTR_SHRA,
+                             CRYPTO_CMD_INSTR_DDATA0TODDATA2);
+        }
+
+        /* DDATA0 = C */
+        crypto->CMD = CRYPTO_CMD_INSTR_DDATA1TODDATA0;
+
+    } /* End of main loop:  while (C != 0)  */
+
+    /* if (D == 1): */
+    /* Decrement D by 1 and test if zero. */
+    CRYPTO_EXECUTE_2(crypto,
+                     CRYPTO_CMD_INSTR_DDATA3TODDATA0,
+                     CRYPTO_CMD_INSTR_DEC);
+
+    if (crypto_ddata0_is_zero(crypto, &status_reg))
+    {
+        CORE_ENTER_CRITICAL();
+        CRYPTO_DDataRead(&crypto->DDATA4, R);
+        CORE_EXIT_CRITICAL();
+    }
+    else
+    {
+        CORE_ENTER_CRITICAL();
+        CRYPTO_DDataWrite(&crypto->DDATA0, N);
+        CORE_EXIT_CRITICAL();
+
+        CRYPTO_EXECUTE_2(crypto,
+                         CRYPTO_CMD_INSTR_SELDDATA0DDATA4,
+                         CRYPTO_CMD_INSTR_SUB
+                         );
+
+        CORE_ENTER_CRITICAL();
+        CRYPTO_DDataRead(&crypto->DDATA0, R);
+        CORE_EXIT_CRITICAL();
+    }
+    return;
+} /* mbedtls_mpi_div_mod  */
+#endif /* MBEDTLS_ECP_NORMALIZE_JAC_MANY_ALT || MBEDTLS_ECP_NORMALIZE_JAC_ALT */
+
+/***************************************************************************//**
+ * @brief
+ *   Enable CRYPTO by setting up control registers for given ecc curve.
+ ******************************************************************************/
+int mbedtls_ecp_device_init( CRYPTO_TypeDef *device, const mbedtls_ecp_group *grp)
```

Comment:
It was considered, but decided against as it would conflict with our atomicity views. If we were to set up the device in internal_ecp_init, then the device would be setup once for the entire chain of crypto operations that constitutes an ECP operation. Our atomicity view is that the device should only be setup once it is actually assigned to an operation, such that we can thread its operation better.

Changed the name of the setup function and made it static.